test/e2e/podAffinityAntiAffinity.go

```diff
@@ -0,0 +1,710 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Pod Affinity and Anti-Affinity", func() {
+
+	// We have two setup podgroups which will create a single task with single pod.
+	// This is required to test the Pod Affinity and Anti-Affinity.
+
+	// since these setup pods will be created ramdomly
+	// we store their node name in these var
+	var setupPodGrpOnesNodeName, setupPodGrpTwosNodeName string
+	setupPodGrpOneName := "setup-pod-one"
+	setupPodGrpTwoName := "setup-pod-two"
+	testPodGrpName := "test-pod"
+
+	// we create and delete these podgroups in the following sub-specs
+	setupPodGrpOne := &jobSpec{
+		name: setupPodGrpOneName,
+		tasks: []taskSpec{
+			{
+				img: "nginx",
+				min: 1,
+				rep: 1,
+				labels: map[string]string{
+					"security": "S1",
+				},
+			},
+		},
+	}
+
+	setupPodGrpTwo := &jobSpec{
+		name: setupPodGrpTwoName,
+		tasks: []taskSpec{
+			{
+				img: "nginx",
```

Comment:
Though it is a legacy, I would suggest we keep all the images consistent, I have seen `"nginx:1.14"` and `nginx`.  

/cc @k82cn @TommyLike

Reply:
This is the same image of 'nginx' img used in other test cases across kube-batch. 
If we finalise on one image we can go and change in all the place in the tests cases.